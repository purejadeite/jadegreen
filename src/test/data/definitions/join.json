{
	"id": "book",
	"sheets": [
		{
			"comment": "結合あり1件読み込み",
			"id": "main",
			"output": true,
			"target": {
			    "name": "bean"
			},
		    "join": {
		    	"sheetId": "sub",
		    	"keyId": "className"
		    },
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "package", "row": 3, "column": 3},
		        {"id": "imports", "row": 4, "column": 3, "splitter": "\n"},
		        {"id": "fields", "row": 8, "breakId": "fieldName", "options": [{"type": "sort", "keyId": "fieldName"}],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "method", "column": 3, "options": [{"type": "uppercamel"}]},
		                {"id": "static", "column": 3, "options": [{"type": "upperunderscore"}]},
		                {"id": "type", "column": 4},
		                {"id": "initial", "column": 5},
		                {"id": "validators", "join": {"tableKeyId": "fieldName"}}
		            ]
		        },
		        {"id": "logicalClassNameValid", "join": {"valueId": "logicalClassName"}}
		    ]
		},
		{
			"comment": "結合先シート",
			"id": "sub",
			"target": {
			    "name": "validation"
			},
		    "cells": [
		        {"id": "logicalClassName", "row": 1, "column": 3},
		        {"id": "className", "row": 2, "column": 3},
		        {"id": "fields", "row": 6, "endKeyId": "fieldName", "options": [{"type": "group", "groupId": "fieldName", "valueId": "validators"}],
		            "columns": [
		                {"id": "logicalFieldName", "column": 2},
		                {"id": "fieldName", "column": 3},
		                {"id": "validationType", "column": 4, "options": [{"type": "mapping", "map": {"notnull": "NotNull"}, "lazy": true}]},
		                {"id": "arg0", "column": 5, "options": [{"type": "tonull"}]},
		                {"id": "arg1", "column": 6, "options": [{"type": "tonull"}]},
		                {"id": "arg2", "column": 7, "options": [{"type": "tonull"}]},
		                {"id": "arg3", "column": 8, "options": [{"type": "tonull"}]}
		            ]
		        }
		    ]
		}
	]
}