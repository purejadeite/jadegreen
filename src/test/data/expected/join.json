[
	{
		"sheetName": "bean",
		"logicalClassName": "シート間の結合",
		"logicalClassNameValid": "シート間の結合(結合先)",
		"className": "Sample",
		"package": "com.purejadeite.generated",
		"imports": [
			"java.io.Serializable",
			"java.util.List",
			"java.util.ArrayList",
			"com.purejadeite.generated.LineBean"
        ],
		"fields": [
			{
				"logicalFieldName": "ファイル有無フラグ",
				"fieldName": "fileExistFlag",
				"method": "FileExistFlag",
				"static": "FILE_EXIST_FLAG",
				"type": "Boolean",
				"initial": "Boolean.TRUE",
				"validators": [
		   			{
						"logicalFieldName": "ファイル有無フラグ",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名",
				"fieldName": "fileName",
				"method": "FileName",
				"static": "FILE_NAME",
				"type": "String",
				"initial": "null",
				"validators": [
		   			{
		   				"logicalFieldName": "ファイル名",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "ファイル名日本語",
				"fieldName": "fileNameJp",
				"method": "FileNameJp",
				"static": "FILE_NAME_JP",
				"type": "String",
				"initial": "null",
				"validators": null
			},
			{
				"logicalFieldName": "ファイルパス",
				"fieldName": "filePath",
				"method": "FilePath",
				"static": "FILE_PATH",
				"type": "String",
				"initial": null,
				"validators": [
					{
						"logicalFieldName": "ファイルパス",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
		   			{
						"logicalFieldName": "ファイルパス",
						"validationType": "length",
						"arg0": "2",
						"arg1": "10",
						"arg2": null,
						"arg3": null
					}
				]
			},
			{
				"logicalFieldName": "行情報",
				"fieldName": "lines",
				"method": "Lines",
				"static": "LINES",
				"type": "List<LineBean>",
				"initial": "new ArrayList<>()",
				"validators": [
		   			{
						"logicalFieldName": "行情報",
						"validationType": "NotNull",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"validationType": "maxlength",
						"arg0": "100",
						"arg1": null,
						"arg2": null,
						"arg3": null
					},
					{
						"logicalFieldName": "行情報",
						"validationType": "kana",
						"arg0": null,
						"arg1": null,
						"arg2": null,
						"arg3": null
					}
				]
			}
		]
	}
]